boskos/boskos_test.go

```diff
@@ -0,0 +1,660 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	"k8s.io/test-infra/boskos/common"
+	"k8s.io/test-infra/boskos/ranch"
+)
+
+func MakeFakeClient(resources []common.Resource) *ranch.Ranch {
+	newRanch := &ranch.Ranch{
+		Resources: resources,
+	}
+
+	return newRanch
+}
+
+func TestAcquire(t *testing.T) {
+	var testcases = []struct {
+		name      string
+		resources []common.Resource
+		path      string
+		code      int
+		method    string
+	}{
+		{
+			name:      "get",
+			resources: []common.Resource{},
+			path:      "?type=t&state=s&owner=o",
+			code:      http.StatusMethodNotAllowed,
+			method:    http.MethodGet,
+		},
+		{
+			name:      "no arg",
+			resources: []common.Resource{},
+			path:      "",
+			code:      http.StatusBadRequest,
+			method:    http.MethodPost,
+		},
+		{
+			name:      "missing type",
+			resources: []common.Resource{},
+			path:      "?state=s&owner=o",
+			code:      http.StatusBadRequest,
+			method:    http.MethodPost,
+		},
+		{
+			name:      "missing state",
+			resources: []common.Resource{},
+			path:      "?type=t&owner=o",
+			code:      http.StatusBadRequest,
+			method:    http.MethodPost,
+		},
+		{
+			name:      "missing owner",
+			resources: []common.Resource{},
+			path:      "?type=t&state=s",
+			code:      http.StatusBadRequest,
+			method:    http.MethodPost,
+		},
+		{
+			name:      "no resource",
```

Comment:
See it is confusing here if you mean "client forgot to include a resource in the request" vs "server cannot find a resource"